test/common/http/http2/http2_connection_fuzz_test.cc

```diff
@@ -0,0 +1,314 @@
+      payload.push_back(static_cast<char>(padding_len));
+      uint32_t stream_dependency = f.stream_dependency();
+      payload.append(reinterpret_cast<char*>(&stream_dependency), sizeof(stream_dependency));
+      deflate_headers(deflater, f.headers(), payload);
+      payload.append(f.padding().c_str(), padding_len);
+    } else if (h2frame.has_priority()) {
+      type = 2;
+      auto f = h2frame.priority();
+      uint32_t stream_dependency = f.stream_dependency();
+      payload.append(reinterpret_cast<char*>(&stream_dependency), sizeof(stream_dependency));
+      payload.push_back(static_cast<char>(f.weight() & 0xff));
+    } else if (h2frame.has_rst()) {
+      type = 3;
+      auto f = h2frame.rst();
+      uint32_t error_code = f.error_code();
+      payload.append(reinterpret_cast<char*>(&error_code), sizeof(error_code));
+    } else if (h2frame.has_settings()) {
+      type = 4;
+      auto f = h2frame.settings();
+      for (auto& setting : f.settings()) {
+        uint16_t id = static_cast<uint16_t>(setting.identifier() & 0xffff);
+        uint32_t value = setting.value();
+        payload.append(reinterpret_cast<char*>(&id), sizeof(id));
+        payload.append(reinterpret_cast<char*>(&value), sizeof(value));
+      }
+    } else if (h2frame.has_push_promise()) {
+      type = 5;
+      auto f = h2frame.push_promise();
+      uint8_t padding_len = static_cast<uint8_t>(f.padding().size() & 0xff);
+      payload.push_back(static_cast<char>(padding_len));
+      uint32_t stream_id = f.stream_id();
+      payload.append(reinterpret_cast<char*>(&stream_id), sizeof(stream_id));
+      deflate_headers(deflater, f.headers(), payload);
+      payload.append(f.padding().c_str(), padding_len);
+    } else if (h2frame.has_ping()) {
+      type = 6;
+      auto f = h2frame.ping();
+      uint64_t pdata = f.data();
+      payload.append(reinterpret_cast<char*>(&pdata), sizeof(pdata));
+    } else if (h2frame.has_go_away()) {
+      type = 7;
+      auto f = h2frame.go_away();
+      uint32_t last_stream_id = f.last_stream_id();
+      payload.append(reinterpret_cast<char*>(&last_stream_id), sizeof(last_stream_id));
+      uint32_t error_code = f.error_code();
+      payload.append(reinterpret_cast<char*>(&error_code), sizeof(error_code));
+      payload.append(f.debug_data());
+    } else if (h2frame.has_window_update()) {
+      type = 8;
+      auto f = h2frame.window_update();
+      uint32_t wsi = f.window_size_increment();
+      payload.append(reinterpret_cast<char*>(&wsi), sizeof(wsi));
+    } else if (h2frame.has_continuation()) {
+      type = 9;
+      auto f = h2frame.continuation();
+      deflate_headers(deflater, f.headers(), payload);
+    } else {
+      // Empty frame
+      return Http2Frame("");
+    }
+
+    Http2Frame result(streamid, type, flags);
+    result.appendData(payload);
+    result.adjustPayloadSize();
+    return result;
+  } else if (pb_frame.has_junk()) {
```

Comment:
Optionally, handle the "junk" case before the "h2frame" case, since it is short and easy to understand.